components/FormFlashcard/FormFlashcard.js

```diff
@@ -1,26 +1,60 @@
 import styled from "styled-components";
 import { RegularButton } from "../Button/Button";
 
-export default function FormFlashcard({ collections, onCreateFlashcard }) {
+export default function FormFlashcard({
+  collections,
+  onCreateFlashcard,
+  headline,
+  isEdit,
+  setIsEdit,
+  currentFlashcard,
+  onEditFlashcard,
+}) {
   function handleSubmit(event) {
     event.preventDefault();
 
     const formData = new FormData(event.target);
     const newFlashcard = Object.fromEntries(formData);
+    if (isEdit) {
+      const updatedFlashcard = {
+        ...newFlashcard,
+        id: currentFlashcard.id,
+        isCorrect: currentFlashcard.isCorrect,
+      };
+      onEditFlashcard(updatedFlashcard);
+    } else {
+      onCreateFlashcard(newFlashcard);
+    }
```

Comment:
Your solution is valid and works well for the current use case. 

However, I would like to draw your attention to the exemplary implementation in the [recap project 4](https://github.com/neuefische/web-react-recap-example/blob/%234-edit-color/src/Components/ColorForm/ColorForm.jsx) 

To make the component simpler and minimize logic, you could consider using a generic onSubmit handler and move the logic to the parent component.